### PR TITLE
feat: refactor table components to functional components [PUL-1298]

### PR DESCRIPTION
**Pull request by mshaaban0**
# Purpose of PR

Refactor table components to functional components, and extend prop types properly. Part of #651

Although we are spreading `{...otherProps}` but ts doesn't really recognise any props the consumer passes that doesn't exist in the props definition. with extending the actual html type it recognises `otherProps` and tight it to the supported attributes

e.g. 
```tsx
export interface TableRowProps extends HTMLProps<HTMLTableRowElement> {
  className?: string;
  testId?: string;
  children: React.ReactNode;
}

// TS works, and recognises onClick
<TableRow onClick={...} /> 


export interface TableRowProps {
  className?: string;
  style?: React.CSSProperties;
  testId?: string;
  children: React.ReactNode;
}
// Won't work
<TableRow onClick={...} /> 
```

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

**Review by denkristoffer — APPROVED** [review]
⚒️

**Review by burakukula — APPROVED** [review]
Looks good! 🥇

File: packages/forma-36-react-components/src/components/Table/Table.tsx [hunk 1]
```diff
@@ -1,36 +1,33 @@
...
 }
 
 const defaultProps: Partial<TableProps> = {
   testId: 'cf-ui-table',
 };
 
-export class Table extends Component<TableProps> {
-  static defaultProps = defaultProps;
+export const Table = (props: TableProps) => {
+  const { className, children, testId, ...otherProps } = props;
```
Comment (gui-santos) [inline packages/forma-36-react-components/src/components/Table/Table.tsx:16]: Wdyt about putting the default props here during this refactor?

**Review by gui-santos — APPROVED** [review]
Amazing 🙌

File: packages/forma-36-react-components/src/components/Table/Table.tsx (hunk 1 above)
Comment (mshaaban0) [reply packages/forma-36-react-components/src/components/Table/Table.tsx:16]: tbh, I don't know the difference, all form controls refactor uses `defaultProps` I'm happy to change if we want it always to be defined inline

Comment (gui-santos) [reply packages/forma-36-react-components/src/components/Table/Table.tsx:16]: One reason to make them inline is that we won't need to do `Partial<TableProps>` for the `defaultProps` anymore

More info here https://github.com/contentful/forma-36/pull/647#discussion_r526932324

Comment (gui-santos) [reply packages/forma-36-react-components/src/components/Table/Table.tsx:16]: It's not urgent tough!

**Pull request merged**